TODO: This patch is necessary because react-native-reanimated breaks SectionList
      (see https://github.com/software-mansion/react-native-reanimated/pull/8301)

diff --git a/node_modules/react-native-reanimated/lib/module/fabricUtils.js b/node_modules/react-native-reanimated/lib/module/fabricUtils.js
index f354459..b85ca20 100644
--- a/node_modules/react-native-reanimated/lib/module/fabricUtils.js
+++ b/node_modules/react-native-reanimated/lib/module/fabricUtils.js
@@ -1,20 +1,22 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 'use strict';
-
+import { ReanimatedError } from './common/errors';
 import { findHostInstance } from './platform-specific/findHostInstance';
-let getInternalInstanceHandleFromPublicInstance;
 export function getShadowNodeWrapperFromRef(ref, hostInstance) {
-  if (getInternalInstanceHandleFromPublicInstance === undefined) {
-    try {
-      getInternalInstanceHandleFromPublicInstance =
-      // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
-      require('react-native/Libraries/ReactNative/ReactFabricPublicInstance/ReactFabricPublicInstance').getInternalInstanceHandleFromPublicInstance ?? (_ref => _ref._internalInstanceHandle);
-    } catch {
-      getInternalInstanceHandleFromPublicInstance = _ref => _ref._internalInstanceHandle;
+  let resolvedInstance =
+    hostInstance?.__internalInstanceHandle ?? ref?.__internalInstanceHandle;
+
+  if (!resolvedInstance) {
+    if (ref.getNativeScrollRef) {
+      resolvedInstance = (ref.getNativeScrollRef() as any)
+        .__internalInstanceHandle;
+    } else if ((ref as any)._reactInternals) {
+      resolvedInstance = findHostInstance(ref).__internalInstanceHandle;
+    } else {
+      throw new ReanimatedError(`Failed to find host instance for a ref.}`);
     }
   }
-  const resolvedRef = ref.getScrollResponder?.()?.getNativeScrollRef?.() ?? ref.getNativeScrollRef?.() ?? ref;
-  const resolvedInstance = ref?.__internalInstanceHandle ?? getInternalInstanceHandleFromPublicInstance(hostInstance ?? findHostInstance(resolvedRef));
-  return resolvedInstance?.stateNode?.node;
+
+  return resolvedInstance!.stateNode.node as ShadowNodeWrapper;
 }
 //# sourceMappingURL=fabricUtils.js.map
diff --git a/node_modules/react-native-reanimated/lib/module/hook/useAnimatedRef.js b/node_modules/react-native-reanimated/lib/module/hook/useAnimatedRef.js
index 8faedaf..fe10a7e 100644
--- a/node_modules/react-native-reanimated/lib/module/hook/useAnimatedRef.js
+++ b/node_modules/react-native-reanimated/lib/module/hook/useAnimatedRef.js
@@ -6,9 +6,6 @@ import { SHOULD_BE_USE_WEB } from '../common/constants';
 import { getShadowNodeWrapperFromRef } from '../fabricUtils';
 import { makeMutable } from '../mutables';
 import { findNodeHandle } from '../platformFunctions/findNodeHandle';
-function getComponentOrScrollable(ref) {
-  return ref.getNativeScrollRef?.() ?? ref.getScrollableNode?.() ?? ref;
-}
 function useAnimatedRefBase(getWrapper) {
   const observers = useRef(new Map()).current;
   const wrapperRef = useRef(null);
@@ -19,7 +16,8 @@ function useAnimatedRefBase(getWrapper) {
         wrapperRef.current = getWrapper(ref);

         // We have to unwrap the tag from the shadow node wrapper.
-        fun.getTag = () => findNodeHandle(getComponentOrScrollable(ref));
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        fun.getTag = () => findNodeHandle(ref);
         fun.current = ref;
         if (observers.size) {
           const currentTag = fun?.getTag?.() ?? null;
@@ -52,7 +50,7 @@ function useAnimatedRefBase(getWrapper) {
 function useAnimatedRefNative() {
   const [sharedWrapper] = useState(() => makeMutable(null));
   const resultRef = useAnimatedRefBase(ref => {
-    const currentWrapper = getShadowNodeWrapperFromRef(getComponentOrScrollable(ref));
+    const currentWrapper = getShadowNodeWrapperFromRef(ref);
     sharedWrapper.value = currentWrapper;
     return currentWrapper;
   });
@@ -69,7 +67,15 @@ function useAnimatedRefNative() {
   return resultRef;
 }
 function useAnimatedRefWeb() {
-  return useAnimatedRefBase(ref => getComponentOrScrollable(ref));
+  return useAnimatedRefBase((ref) => {
+    if (ref.getScrollableNode) {
+      return ref.getScrollableNode();
+    }
+    if (ref.getNativeScrollRef) {
+      return ref.getNativeScrollRef();
+    }
+    return ref;
+  });
 }

 /**
diff --git a/node_modules/react-native-reanimated/src/commonTypes.ts b/node_modules/react-native-reanimated/src/commonTypes.ts
index d57914f..6fd5cce 100644
--- a/node_modules/react-native-reanimated/src/commonTypes.ts
+++ b/node_modules/react-native-reanimated/src/commonTypes.ts
@@ -360,7 +360,7 @@ export enum InterfaceOrientation {
 }

 export type ShadowNodeWrapper = {
-  __hostObjectShadowNodeWrapper: never;
+  __nativeStateShadowNodeWrapper: never;
 };

 export enum KeyboardState {
diff --git a/node_modules/react-native-reanimated/src/fabricUtils.ts b/node_modules/react-native-reanimated/src/fabricUtils.ts
index 3661747..811e357 100644
--- a/node_modules/react-native-reanimated/src/fabricUtils.ts
+++ b/node_modules/react-native-reanimated/src/fabricUtils.ts
@@ -1,40 +1,27 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 'use strict';
+import { ReanimatedError } from './common/errors';
 import type { ShadowNodeWrapper, WrapperRef } from './commonTypes';
 import type { HostInstance } from './platform-specific/findHostInstance';
 import { findHostInstance } from './platform-specific/findHostInstance';

-let getInternalInstanceHandleFromPublicInstance: (ref: unknown) => {
-  stateNode: { node: unknown };
-};
-
 export function getShadowNodeWrapperFromRef(
   ref: WrapperRef,
   hostInstance?: HostInstance
 ): ShadowNodeWrapper {
-  if (getInternalInstanceHandleFromPublicInstance === undefined) {
-    try {
-      getInternalInstanceHandleFromPublicInstance =
-        // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
-        require('react-native/Libraries/ReactNative/ReactFabricPublicInstance/ReactFabricPublicInstance')
-          .getInternalInstanceHandleFromPublicInstance ??
-        ((_ref: any) => _ref._internalInstanceHandle);
-    } catch {
-      getInternalInstanceHandleFromPublicInstance = (_ref: any) =>
-        _ref._internalInstanceHandle;
+  let resolvedInstance =
+    hostInstance?.__internalInstanceHandle ?? ref?.__internalInstanceHandle;
+
+  if (!resolvedInstance) {
+    if (ref.getNativeScrollRef) {
+      resolvedInstance = (ref.getNativeScrollRef() as any)
+        .__internalInstanceHandle;
+    } else if ((ref as any)._reactInternals) {
+        resolvedInstance = findHostInstance(ref).__internalInstanceHandle;
+    } else {
+        throw new ReanimatedError(`Failed to find host instance for a ref.}`);
     }
   }

-  const resolvedRef =
-    ref.getScrollResponder?.()?.getNativeScrollRef?.() ??
-    ref.getNativeScrollRef?.() ??
-    ref;
-
-  const resolvedInstance =
-    ref?.__internalInstanceHandle ??
-    getInternalInstanceHandleFromPublicInstance(
-      hostInstance ?? findHostInstance(resolvedRef)
-    );
-
-  return resolvedInstance?.stateNode?.node;
+  return resolvedInstance!.stateNode.node as ShadowNodeWrapper;
 }
diff --git a/node_modules/react-native-reanimated/src/hook/useAnimatedRef.ts b/node_modules/react-native-reanimated/src/hook/useAnimatedRef.ts
index ad903a6..7696c2f 100644
--- a/node_modules/react-native-reanimated/src/hook/useAnimatedRef.ts
+++ b/node_modules/react-native-reanimated/src/hook/useAnimatedRef.ts
@@ -17,10 +17,6 @@ import type {
   MaybeObserverCleanup,
 } from './commonTypes';

-function getComponentOrScrollable(ref: WrapperRef) {
-  return ref.getNativeScrollRef?.() ?? ref.getScrollableNode?.() ?? ref;
-}
-
 function useAnimatedRefBase<TRef extends WrapperRef>(
   getWrapper: (ref: TRef) => ShadowNodeWrapper
 ): AnimatedRef<TRef> {
@@ -36,7 +32,8 @@ function useAnimatedRefBase<TRef extends WrapperRef>(
         wrapperRef.current = getWrapper(ref);

         // We have to unwrap the tag from the shadow node wrapper.
-        fun.getTag = () => findNodeHandle(getComponentOrScrollable(ref));
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        fun.getTag = () => findNodeHandle(ref as any);
         fun.current = ref;

         if (observers.size) {
@@ -81,9 +78,7 @@ function useAnimatedRefNative<
   );

   const resultRef = useAnimatedRefBase<TRef>((ref) => {
-    const currentWrapper = getShadowNodeWrapperFromRef(
-      getComponentOrScrollable(ref)
-    );
+    const currentWrapper = getShadowNodeWrapperFromRef(ref);

     sharedWrapper.value = currentWrapper;

@@ -106,7 +101,15 @@ function useAnimatedRefNative<
 function useAnimatedRefWeb<
   TRef extends WrapperRef = React.Component,
 >(): AnimatedRef<TRef> {
-  return useAnimatedRefBase<TRef>((ref) => getComponentOrScrollable(ref));
+  return useAnimatedRefBase<TRef>((ref) => {
+    if (ref.getScrollableNode) {
+      return ref.getScrollableNode();
+    }
+    if (ref.getNativeScrollRef) {
+      return ref.getNativeScrollRef();
+    }
+    return ref;
+  });
 }

 /**
